de4a-connector/src/main/java/eu/de4a/connector/service/spring/Conf.java

```diff
@@ -201,22 +205,30 @@ public RestTemplate restTemplate() {
 	public HttpClient httpClient() {
 		try {
 			LOG.debug("SSL context setted to: {}", sslContextEnabled);
+			SSLConnectionSocketFactory factory;
 			if(Boolean.TRUE.toString().equals(sslContextEnabled)) {
-				SSLConnectionSocketFactory factory = sslConnectionSocketFactory();
-				return HttpClientBuilder.create().setSSLSocketFactory(factory).build();
+				factory = new SSLConnectionSocketFactory(sslContext());
+				
 			} else {
-				return HttpClientBuilder.create().build();
+				factory = new SSLConnectionSocketFactory(sslContextTrustAll());
 			}
+			return HttpClientBuilder.create().setSSLSocketFactory(factory).build();
 		} catch (Exception e) {
 			LOG.error("Unable to create SSL factory", e);
 		}
 		return HttpClientBuilder.create().build();
 
 	}
-
-	public SSLConnectionSocketFactory sslConnectionSocketFactory() {
-		SSLContext context = sslContext();
-		return new SSLConnectionSocketFactory(context);
+	
+	public SSLContext sslContextTrustAll() {
+		TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
+		try {
+			return SSLContexts.custom().loadTrustMaterial(null, acceptingTrustStrategy)
+				.build();
+		} catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
+			LOG.error("There was a problem creating sslContextTrustAll", e);
+			return null;
```

Comment:
I suggest to throw an IllegalStateException here, because this is a security critical incident